src/wire/ip.rs

```diff
@@ -496,75 +496,88 @@ impl Repr {
         match self {
             #[cfg(feature = "proto-ipv4")]
             &Repr::Unspecified {
-                src_addr: Address::Ipv4(src_addr),
+                src_addr: src_addr @ Address::Unspecified,
                 dst_addr: Address::Ipv4(dst_addr),
                 protocol, payload_len, hop_limit
-            } => {
+            } |
+            &Repr::Unspecified {
```

Comment:
Shouldn't you remove the `resolve_unspecified` macro above and the wildcard cases below? Or am I missing something?

Reply:
No, it's used below for the `Repr::Ipv4` and `Repr::Ipv6` cases.

Reply:
Since the branches in my case are equivalent, I could write a similar macro if you think that would be better. I didn't because `Ipv6Repr` and `Ipv4Repr` differ at `proto`/`next_header` so the macro would be somewhat messy.